libcloud/utils/misc.py

```diff
@@ -282,3 +292,28 @@ def __repr__(self):
 
     def __str__(self):
         return str(self.__repr__())
+
+
+def retry(timeout, retry_delay, backoff, func, *args, **kwargs):
+    """
+    Retry decorator helps to handle common exception.
+    """
+    timeout = DEFAULT_TIMEOUT if timeout is None else timeout
+    bsleep = DEFAULT_SLEEP if retry_delay is None else retry_delay
+    backoff = DEFAULT_BACKCOFF if backoff is None else backoff
+    end = datetime.now() + timedelta(seconds=timeout)
+    exc_info = None
+    while datetime.now() < end:
+        try:
+            result = func(*args, **kwargs)
+            return result
+        except RateLimit as exc:
```

Comment:
Since the function lives in the `misc` module and it's already "kinda" generic it would make sense if user could also pass in an exception (or maybe a callable) upon which we should retry.

For example, an argument called "`retry_exception=RateLimit` or something along those lines.